xarray/core/groupby.py

```diff
@@ -370,8 +370,11 @@ def __init__(
                 group = group.dropna(group_dim)
 
             # look through group to find the unique values
+            sort = bins is None and (
+                not isinstance(safe_cast_to_index(group), pd.MultiIndex)
```

Comment:
One performance opt is to only do `safe_cast_to_index(group)` once; though I don't think it's huge

Reply:
Sure, done in 4407f40.